ChessDotNet.Variants/Horde/HordeChessGame.cs

```diff
@@ -61,7 +63,19 @@ public HordeChessGame() : base()
         public HordeChessGame(Piece[][] board, Player whoseTurn) : base(board, whoseTurn) { }
         public HordeChessGame(GameCreationData data) : base(data) { }
         public HordeChessGame(string fen) : base(fen) { }
-        public HordeChessGame(IEnumerable<Move> moves, bool movesAreValidated) : base(moves, movesAreValidated) { }
+        public HordeChessGame(IEnumerable<Move> moves, bool movesAreValidated) : this() {
+            if (moves == null)
+                throw new ArgumentNullException("moves");
+            if (moves.Count() == 0)
```

Comment:
Why can't `moves.Count()` be zero? It may not seem to make much sense, but there's certainly nothing wrong with it and nothing that breaks as a result.

Reply:
Just to preserve the behaviour as in ChessGame constructor